design-proposals/emf-upgrade-steps.md

```diff
@@ -0,0 +1,74 @@
+# Design Proposal: EMF upgrade steps
+
+Author(s): Diwan Chandrabose, Sunil Parida
+
+Last updated: 2025-07-07
+
+## Abstract
+
+This proposal captures EMF upgrade paths and the steps involved in the upgrades.
+This document has to be reviewed as part of every planned release
+to make sure upgrade paths to the new release will be designed and supported.
+
+## Upgrades
+
+### Upgrade from EMF 3.0 to 3.1
+
+#### Disruptive upgrade
+
+In EMF 3.1, the node architecture is being changed to install and run a k3s cluster in edges instead of rke2.
+Any edge node onboarded on EMF 3.1 (or later versions in the future) and EMF 3.0 will  
+run k3s cluster and rke2 cluster respectively.
+When an EMF 3.0 instance has to be upgraded to 3.1,
+the existing 3.0 edge nodes will need to be re-onboarded post EMF upgrade.
+Edge node operators must make sure to manually back up any application manifests,
+EMF deployment packages and data on the nodes before commencing the EMF 3.0 to 3.1 upgrade.
+And manually restore the apps and data after EMF upgrade and re-onboarding the nodes.
+Future upgrades from 3.1 to 3.2 or later will be non-disruptive
+as we dont plan to include breaking changes in the edge node architecture in 3.2.
+
+Any effort to provide an automated edge node rke2 to k3s cluster replacement
+in an attempt to build a complete non-disruptive upgrade,
+would jeopardize 3.1 release.
+More importantly the effort would also be useful only one time
+to upgrade any existing 3.0 instance.
+3.1 to 3.2 and later upgrades would not require this approach.
+
+#### Upgrade steps
+
+#### Step 1: Delete existing nodes in 3.0
+
+Do the following for each host pertaining to each edge node onboarded in EMF.
+De-authorizing and deleting a host does not wipe out any deployment packages that are
+deployed in the edge cluster.
+The deployment packages can be applied to the host once they are re-onboarded and re-provisioned.
+
+- Manually back up any desired apps data in the edge node/cluster
+- De-authorize the host by following the steps given in the doc [here][De-auth Host Documentation]
+- Delete the host by following the steps given in the doc [here][Delete Host Documentation]
+
```

Comment:
this is not ok - we have to instruct people for first remove the edge cluster from CO. Otherwise you have to document the forcefully removal post de-auth/delete.

Reply:
Looking at the doc, https://docs.openedgeplatform.intel.com/edge-manage-docs/dev/user_guide/set_up_edge_infra/delete_host.html, I thought direct delete of host is taken care. If thats correct, it would save steps and simplify the pre-upgrade procedure. Are there more details here such as single node vs multi node cluster etc?

> You can delete a Host that has a deployment associated to it. In such cases, the deployment will continue to be reconciled within the cluster and remain operational.

Reply:
@hyunsun I am adding these steps. for step # 3 in pre-upgrade to remove base extns, could you point me to any existing doc?